Google cloud interconnect

privateip opened the pull request:
This change updates the google_cloud_interconnect role to focus on the
minimimum required configuration for creating google cloud
interconnects.  This change now requires passing `gcp_network` into the
role in order to configure the interconnect.

Commit:
removes extra provisioning steps

This change updates the google_cloud_interconnect role to focus on the
minimimum required configuration for creating google cloud
interconnects.  This change now requires passing `gcp_network` into the
role in order to configure the interconnect.

Commit:
adds additional check for selfLink

mtraynham reviewed (commented):
Code looks good, but following along with the AWS PR, did you want include the vpc variable in the documentation?

Commit:
generate dynamic names for all resources

This commit removes the defaults and now generates a dynamic name for
all configured resources.  It introduces a new control variable called
`fabric` which can be used for attaching this connection to a Pureport
Multicloud Fabric implementation.

Commit:
updates google_cloud_interconnect role for easier use

This commit makes significant changes to the google_cloud_interconnect
role such that it can now create connections with minimal vlaues.  The
only required values for using this role are `pureport_connection_speed`
and `gcp_region`.  All other values are determined automatically based
on the value of `fabric`.  If no value is defined for `fabric`, then a
default value is assigned.

Commit:
validates the value for state

Commit:
update  logic for generating pureport connection name
